EIPS/eip-7873.md

```diff
@@ -147,85 +145,19 @@ Introduce a new instruction on the same block number [EIP-3540](./eip-3540.md) i
-
-```solidity
-{
-    /// Takes [tx_initcode_hash][salt][unsafe_flag][init_data] as input,
-    /// creates contract and returns the address or failure otherwise.
-
-    /// tx_initcode_hash and salt are 32 bytes wide, unsafe_flag is 1 byte wide, init_data can be any width.
-    /// If unsafe_flag is 0x00, `CALLER` is included in the salt and impacts the target address of the created account.
-    /// If unsafe_flag is non-zero, `CALLER` is excluded, with 32 zero-bytes used instead. Deployment by any sender will result in same target address. 
-    /// Deployments using non-zero unsafe_flag are susceptible to front-running.
-
-    // init_data.length can be 0, but the first 65 bytes are mandatory
-    let size := calldatasize()
-    if lt(size, 65) { revert(0, 0) }
-
-    // copy tx_initcode_hash and salt to memory to hash.
-    // note that the unsafe_flag doesn't need to be included. If set to non-zero, 32 zero-bytes
-    // are used instead of the CALLER address bytes, which prevents hash collisions between
-    // deployments using unsafe_flag and those not using it.
-    calldatacopy(0, 0, 64)
-
-    // mask out the 31 bytes that follow the flag in input data
-    let unsafe_flag := byte(0, calldataload(64))
-
-    if iszero(unsafe_flag) {
-        // store caller in memory to hash, just after salt
-        mstore(64, caller())
-    } else {
-        // exclude sender from salt - susceptible to front-running
-        // assuming no bytes were written to memory word at 64 and it's all zeros
-    }
-
-    let init_data_size := sub(size, 65)
-
-    // copy init_data to memory to hash, just after caller (or zeros)
-    calldatacopy(96, 65, init_data_size)
-
-    // final_salt = keccak256(tx_initcode_hash | salt | caller_or_zeros | init_data)
-    let final_salt := keccak256(0, 96 + init_data_size)
-
-    let tx_initcode_hash := calldataload(0)
-
-    // reuse init_data which has been already copied to memory above
-    let ret := txcreate(tx_initcode_hash, callvalue(), final_salt, 96, init_data_size)
-
-    if iszero(ret) {
-        let ret_data_size := returndatasize()
-        returndatacopy(0, 0, ret_data_size)
-        revert(0, ret_data_size)
-    }
-
-    mstore(0, ret)
-    return(0, 32)
+## Rationale
 
-    // Helper to compile this with existing Solidity (with --strict-assembly mode)
-    function txcreate(a, b, c, d, e) -> f {
-        f := verbatim_5i_1o(hex"ed", a, b, c, d, e)
-    }
-}
-```
+### `to: nil` transactions cannot use `TXCREATE`
 
-## Rationale
+By ruling out `InitcodeTransactions` which have `to: nil` we're ruling out the possibility of making such transactions with a legacy EVM initcode which only calls into `TXCREATE` and `SELFDESTRUCTS`. Since the main purpose of allowing `TXCREATE` in legacy EVM code is to bootstrap proper EOF factories, the constrained rules make more sense.
```

Comment:
This doesn't really make sense to me, `InitcodeTransactions` just don't have a field for code other than `initcodes`, `data` is only used for input.

Reply:
Or this should describe it from another angle: why such hack of code into `data` is not needed. But I would just drop this section.